src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/eventstream/EventStreamAdherenceReportGenerator.java

```diff
@@ -0,0 +1,229 @@
+package org.sagebionetworks.bridge.models.schedules2.adherence.eventstream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.toMap;
+import static org.sagebionetworks.bridge.models.schedules2.adherence.AdherenceUtils.calculateSessionState;
+import static org.sagebionetworks.bridge.models.schedules2.adherence.SessionCompletionState.COMPLIANT;
+import static org.sagebionetworks.bridge.models.schedules2.adherence.SessionCompletionState.NONCOMPLIANT;
+import static org.sagebionetworks.bridge.models.schedules2.adherence.SessionCompletionState.UNKNOWN;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Days;
+import org.joda.time.LocalDate;
+import org.sagebionetworks.bridge.models.activities.StudyActivityEvent;
+import org.sagebionetworks.bridge.models.schedules2.adherence.AdherenceRecord;
+import org.sagebionetworks.bridge.models.schedules2.adherence.SessionCompletionState;
+import org.sagebionetworks.bridge.models.schedules2.timelines.TimelineMetadata;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+public class EventStreamAdherenceReportGenerator {
+
+    private final boolean showActive;
+    private final DateTime now;
+    private final String clientTimeZone;
+    private final List<TimelineMetadata> metadata;
+    private final Map<String, EventStream> streamsByEventId;
+    private final Map<String, EventStreamDay> streamsByStreamKey;
+    private final Map<String, AdherenceRecord> adherenceByEventId;
+    private final Map<String, Integer> daysSinceEventByEventId;
+    private final Map<String, DateTime> eventTimestampByEventId;
+
+    public EventStreamAdherenceReportGenerator(EventStreamAdherenceReportGenerator.Builder builder) {
+        showActive = builder.showActiveOnly;
+        clientTimeZone = builder.clientTimeZone;
+        now = builder.now;
+        metadata = builder.metadata;
+        streamsByEventId = new HashMap<>();
+        daysSinceEventByEventId = new HashMap<>();
+        eventTimestampByEventId = new HashMap<>();
+        streamsByStreamKey = new HashMap<>();
+        adherenceByEventId = builder.adherenceRecords.stream()
+                .collect(toMap(AdherenceRecord::getInstanceGuid, (a) -> a));
+        
+        LocalDate localNow = (builder.now != null) ? builder.now.toLocalDate() : null;
+        for (StudyActivityEvent event : builder.events) {
+            DateTime eventTimestamp = event.getTimestamp();
+            if (event.getClientTimeZone() != null) {
+                eventTimestamp = eventTimestamp.withZone(DateTimeZone.forID(event.getClientTimeZone()));
+            } else {
+                eventTimestamp = eventTimestamp.withZone(builder.now.getZone());
```

Comment:
Should we directly use the clientTimeZone being passed in?

Reply:
checks event zone, participant's persisted zone, and the zone of "now" in that order. Tests added for this.